sql/parser/grant.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+// This code was derived from https://github.com/youtube/vitess.
+//
+// Copyright 2012, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package parser
+
+import "fmt"
+
+// Grant represents a GRANT statement.
+type Grant struct {
+	Privileges []string
+	Target     Name
+	Grantees   []string
+}
+
+func (node *Grant) String() string {
+	return fmt.Sprintf("GRANT %s ON %v TO %v", node.Privileges, node.Target, node.Grantees)
```

Comment:
Will fix the output. How similar do we want it to be? Should it generate exactly the same string?

The grantee is a "name" for now, which is rather strict, and will definitely need to change (see TODO on sql.y, grantee_list). We should figure out what is allowed for usernames. This will be applied in a lot of places.